Blog for Rust 1.49.0 (#745)
rust-lang/blog.rust-lang.org: posts/2020-12-31-Rust-1.49.0.md

---
layout: post

> title: "Announcing Rust 1.40.0"

Reviewer: ```suggestion
title: "Announcing Rust 1.49.0"
```

Author: I checked FIVE TIMES, hahaha.

Reviewer 2: Pushed a commit fixing this.

author: The Rust Release Team
release: true
---

The Rust team is happy to announce a new version of Rust, 1.49.0. Rust is a
programming language that is empowering everyone to build reliable and
efficient software.

If you have a previous version of Rust installed via rustup, getting Rust
1.49.0 is as easy as:

```console
rustup update stable
```

If you don't have it already, you can [get `rustup`][install] from the
appropriate page on our website, and check out the [detailed release notes for
1.49.0][notes] on GitHub.

[install]: https://www.rust-lang.org/install.html
[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1490-2020-12-31

## What's in 1.49.0 stable

For this release, we have some new targets and an improvement to the test
framework. See the [detailed release notes][notes] to learn about other
changes not covered by this post.

> ### aarch64 targets

Reviewer 2: Hmm, I feel like we should expand this. Having the first non-x86 platform reaching Tier 1 is a huge deal, and a ton of people are interested in support for the new Apple M1. Possible sketch of what content to put in:

* 64-bit ARM on Linux reached Tier 1 support
  * Every change done to the compiler now requires the full test suite to pass on the target.
  * First non-x86 target to be promoted to Tier 1!
  * Maybe: where the target could be used
* 64-bit ARM on Windows and Apple Silicon reached Tier 2 support!
  * Pre-built compilers are now guaranteed to be available through rustup
  * We're not running the compiler's test suite on those platforms, so there could be bugs in the compiler
  * Maybe: where the targets could be used

Author: You had mentioned really wanting these to be two sections, is there a reason why? they feel like the same thing to me, though I do think expanding on it would be good.

Reviewer 2: Mostly because the guarantees provided by Tier 1 and Tier 2 are so different that merging them together might confuse people. Also I kinda want to give the first non-x86 target the spotlight on its own :D

---
layout: post
title: "Announcing Rust 1.49.0"
author: The Rust Release Team
release: true
---

The Rust team is happy to announce a new version of Rust, 1.49.0. Rust is a
programming language that is empowering everyone to build reliable and
efficient software.

If you have a previous version of Rust installed via rustup, getting Rust
1.49.0 is as easy as:

```console
rustup update stable
```

If you don't have it already, you can [get `rustup`][install] from the
appropriate page on our website, and check out the [detailed release notes for
1.49.0][notes] on GitHub.

[install]: https://www.rust-lang.org/install.html
[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1490-2020-12-31

## What's in 1.49.0 stable

For this release, we have some new targets and an improvement to the test
framework. See the [detailed release notes][notes] to learn about other
changes not covered by this post.

### 64-bit ARM Linux reaches Tier 1

The Rust compiler supports [a wide variety of targets][platform-support], but
the Rust Team can't provide the same level of support for all of them. To

> clearly mark how supported each target is, we created a tiering system:

Reviewer 3: Nit: we haven't created the system this release. Perhaps "we utilize a tiering system" would be a better way to phrase this?

* Tier 3 targets are technically supported by the compiler, but we don't check
  whether their code build or passes the tests, and we don't provide any
  prebuilt binaries as part of our releases.
* Tier 2 targets are guaranteed to build and we provide prebuilt binaries, but
  we don't execute the test suite on those platforms: the produced binaries
  might not work or might have bugs.
* Tier 1 targets provide the highest support guarantee, and we run the full
  suite on those platforms for every change merged in the compiler. Prebuilt
  binaries are also available.

Rust 1.49.0 promotes the `aarch64-unknown-linux-gnu` target to Tier 1 support,
bringing our highest guarantees to users of 64-bit ARM systems running Linux!
We expect this change to benefit workloads spanning from embedded to desktops
and servers.

This is an important milestone for the project, since it's the first time a

> non-x86 target reaches Tier 1 support: we hope this will pave the way for more

Reviewer 4: ```suggestion
non-x86 target has reached Tier 1 support: we hope this will pave the way for more
```

Reviewer 2: Fixed.

more targets to reach our highest tier in the future.

Note that Android is not affected by this change as it uses a different Tier 2
target.

[platform-support]: https://doc.rust-lang.org/stable/rustc/platform-support.html

### 64-bit ARM macOS and Windows reach Tier 2

Rust 1.49.0 also features two targets reaching Tier 2 support:

* The `aarch64-apple-darwin` target brings support for Rust on Apple M1 systems.

> * The `aarch64-pc-windows-mvc` target brings support for Rust on 64-bit ARM

Reviewer 3: msvc

---
layout: post
title: "Announcing Rust 1.49.0"
author: The Rust Release Team
release: true
---

The Rust team is happy to announce a new version of Rust, 1.49.0. Rust is a
programming language that is empowering everyone to build reliable and
efficient software.

If you have a previous version of Rust installed via rustup, getting Rust
1.49.0 is as easy as:

```console
rustup update stable
```

If you don't have it already, you can [get `rustup`][install] from the
appropriate page on our website, and check out the [detailed release notes for
1.49.0][notes] on GitHub.

[install]: https://www.rust-lang.org/install.html
[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1490-2020-12-31

## What's in 1.49.0 stable

For this release, we have some new targets and an improvement to the test
framework. See the [detailed release notes][notes] to learn about other
changes not covered by this post.

### 64-bit ARM Linux reaches Tier 1

The Rust compiler supports [a wide variety of targets][platform-support], but
the Rust Team can't provide the same level of support for all of them. To

> clearly mark how supported each target is we created a tiering system:

Reviewer 5: ```suggestion
clearly mark how supported each target is, we created a tiering system:
```

* Tier 3 targets are technically supported by the compiler, but we don't check
  whether their code build or passes the tests, and we don't provide any
  prebuilt binaries as part of our releases.
* Tier 2 targets are guaranteed to build and we provide prebuilt binaries, but
  we don't execute the test suite on those platforms: the produced binaries
  might not work or might have bugs.
* Tier 1 targets provide the highest support guarantee, and we run the full
  suite on those platforms for every change merged in the compiler. Prebuilt

>   binaries are also available.

Reviewer 5: This is already in the platform-support page, right? Does it make sense to repeat it in the blog? I think the following paragraph ('highest guarantees') is good enough to understand why this is a big deal.

Reviewer 2: I'd prefer to keep it to provide more context for people who are not already familiar with our policy and who are going to miss the link.

Rust 1.49.0 promotes the `aarch64-unknown-linux-gnu` target to Tier 1 support,
bringing our highest guarantees to users of 64-bit ARM systems running Linux!
We expect this change to benefit workloads spanning from embedded to desktops
and servers.

This is an important milestone for the project: it's the first time a non-x86

> target reaches the highest level of Quality Assurance and support provided by

Reviewer 5: ```suggestion
target reaches the highest level of support provided by
```

> target reaches the highest level of Quality Assurance and support provided by

Reviewer 4: I think my only nit is that we currently still limit crater to x86_64-unknown-linux-gnu; maybe we can just say "It's the first non-x86 target that reaches tier 1"?
